nemo/core/optim/lr_scheduler.py

```diff
@@ -665,6 +665,171 @@ def _get_lr(self, step):
         return new_lrs
 
 
+class MegatronAnnealing(_LRScheduler):
+    """Meagton learning rate and weight decay Anneals"""
+
+    def __init__(
+        self,
+        optimizer,
+        *,
+        init_lr,
+        max_lr,
+        min_lr,
+        lr_warmup_steps,
+        lr_decay_steps,
+        lr_decay_style,
+        start_wd,
+        end_wd,
+        wd_incr_steps,
+        wd_incr_style,
+        use_checkpoint_opt_param_scheduler=True,
+        override_opt_param_scheduler=False,
+        last_epoch=-1,
+    ):
+        """
```

Comment:
I think the rest of the classes put the doc-string right below `class MegatronAnnealing`